The public link edit template was outside public upload check

sharidas opened the pull request:
The public link edit template was outside public upload check.
And hence for creating the public link for files the full
permission option was shown. This change tries to address
the issue.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" in case the PR still has open tasks
- Set label "backport-request" if backport is needed
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->
The `Download / View / Edit` option was outside the check for variable `publicUploadPossible` in the template. And this caused the issue for files. When user tries to create the public links for files the `Download / View / Edit` option is visible. But when user clicks it only view permission is available for the public link created. So in this change set the `Download / View / Edit` option is added inside the `publicUploadPossible` check and hence the consistency is brought back for creating public links for files.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes https://github.com/owncloud/core/issues/35223

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
The `Download / View / Edit` option was moved out of the `publicUploadPossible` check in the template. Hence there caused inconistency when the public links for files created.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- Here is the simple screen cast 
![upload](https://user-images.githubusercontent.com/3600427/57704237-224e6d00-767f-11e9-81b3-95ea36c2ca7f.gif)
- [x] Tested with this change that permission created by public links for files is `1`.
- [x] Tested public links for `Download / View / Upload` works as it is.
- [x] Tested public links created with permission `Download / View / Edit` works.
- [x] Tested switching permissions for the public links when created for folder is not affected with this change.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

PVince81 commented:
JS unit test would be nice, I think we have existing coverage there

sharidas commented:
I too expected the coverage at https://github.com/owncloud/core/blob/master/core/js/tests/specs/sharedialoglinklistviewSpec.js or https://github.com/owncloud/core/blob/master/core/js/tests/specs/sharedialoglinkshareviewSpec.js.

sharidas commented:
Let me know if this looks ok.

PVince81 commented:
@sharidas can you extend the tests ?

sharidas commented:
Updated this PR with a minor js test.

PVince81 commented on core/js/tests/specs/sharedialoglinklistviewSpec.js:
```diff
@@ -160,6 +160,33 @@ describe('OC.Share.ShareDialogLinkListView', function() {
 			expect(popup.itemModel).toEqual(itemModel);
 			defaultDateStub.restore();
 		});
+		it('when file is clicked to create public link it sets default values with read only permission', function() {
```
the sentence must start with "it". this is why the framework's test method is called `it`.

so: `it('sets default values ...')`

(optional change, keep in mind for next time)

PVince81 commented on core/js/tests/specs/sharedialoglinklistviewSpec.js:
```diff
@@ -160,6 +160,33 @@ describe('OC.Share.ShareDialogLinkListView', function() {
+
+			var popup = showPopup();
+
+			expect(popup.model.toJSON()).toEqual({
```
the test doesn't seem to cover the fact that the two options are available for folders and only one for file

is this covered in other places ?

sharidas replied:
I have updated the test. Yes it is covered in a different location https://github.com/owncloud/core/blob/master/core/js/tests/specs/sharedialoglinkshareviewSpec.js

sharidas replied:
imho, the updated test looks good to me, compared to the previous version of test I wrote.

sharidas committed:
The public link edit template was outside public upload check

The public link edit template was outside public upload check.
And hence for creating the public link for files the full
permission option was shown. This change tries to address
the issue.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

PVince81 commented:
some test failures ?

PVince81 reviewed (approved):
:+1:

PVince81 commented:
@sharidas if you are confident that CI will pass, please backport to release-10.2.0 right away

sharidas commented:
Backport to release branch https://github.com/owncloud/core/pull/35238

PVince81 reviewed (approved):
:+1:

PVince81 reviewed (approved):
:+1:
